### PR TITLE
Refactor/kws 1796

### PR DESCRIPTION
**Pull request by GuilhermeMota93**

### CODE REVIEW

File: kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/responses/BaseAuthResponse.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package kws.superawesome.tv.kwssdk.base.responses
+
+import com.google.gson.annotations.SerializedName
+
+/**
+ * Created by guilherme.mota on 08/12/2017.
+ */
+data class BaseAuthResponse(
+        @SerializedName("access_token")  val sessionToken: String? = null)
```
Comment (devgabrielcoman) [inline kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/responses/BaseAuthResponse.kt]: Should probably call this class "AuthResponse".
Should probably call "sessionToken" just "token"

File: kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/requests/LoginUserRequest.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * Created by guilherme.mota on 08/12/2017.
+ */
+internal class LoginUserRequest(environment: NetworkEnvironment,
+                                username: String,
+                                password: String,
+                                clientID: String,
+                                clientSecret: String
+) :
+        BaseRequest(environment = environment) {
+    override val mediaType: String
```
Comment (devgabrielcoman) [inline kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/requests/LoginUserRequest.kt]: can replace this with
override val mediaType: String = "application/x-www-form-urlencoded; charset=utf-8"

No need for the "get() =  ...."

File: kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/requests/LoginUserRequest.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        get() = "application/x-www-form-urlencoded; charset=utf-8"
+    override val isURLEncoded: Boolean
```
Comment (devgabrielcoman) [inline kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/requests/LoginUserRequest.kt]: Why is this needed anymore? The "LoginUserRequest" will work with a "NetworkUrlEncodedTask", no?

File: kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/providers/LoginProvider.kt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+            // network success case
+            if (rawString != null && networkError == null) {
+
+                val parseRequest = ParseJsonRequest(rawString = rawString)
+                val parseTask = ParseJsonTask()
+                val authResponse = parseTask.execute<BaseAuthResponse>(input = parseRequest) ?: BaseAuthResponse()
+                var token = authResponse.sessionToken
+
+                token?.let {
```
Comment (devgabrielcoman) [inline kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/providers/LoginProvider.kt]: I would replace this with if (token != null) { .... } else { .... }

The syntax when trying to do this with var?.let { ... } :? ... is horrendous

File: baselib/src/main/java/kws/superawesome/tv/androidbaselib/network/BaseNetworkTask.kt [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+ */
+abstract class BaseNetworkTask : AsyncTask<NetworkRequest, String?> {
+
+    override fun execute(input: NetworkRequest, callback: (String?, Throwable?) -> Unit) {
+
+        val client = OkHttpClient()
+
+        val method = input.method
+
+        val queryString = formQueryString(query = input.parameters)
```
Comment (devgabrielcoman) [inline baselib/src/main/java/kws/superawesome/tv/androidbaselib/network/BaseNetworkTask.kt]: put this back in!

File: baselib/src/main/java/kws/superawesome/tv/androidbaselib/network/BaseNetworkTask.kt [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        val domain = input.environment.domain
+        val endpoint = input.endpoint
+        val url = "$domain$endpoint"
```
Comment (devgabrielcoman) [inline baselib/src/main/java/kws/superawesome/tv/androidbaselib/network/BaseNetworkTask.kt]: this should be:
val url = "$domain$endpoint$queryString"

File: kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/responses/BaseAuthResponse.kt (hunk 1 above)
Comment (GuilhermeMota93) [reply kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/responses/BaseAuthResponse.kt]: Will rename "AuthResponse" to "LoginResponse" as the response for the Create user is different.
